### PR TITLE
Fix Woopay Button Logo Cut Off in Certain Views

### LINKED ISSUE #6263: WooPay Button Logo cut off in certain views
### Describe the bug

This issue was noticed while viewing the WooPay Express Checkout button on the blocks cart with the base storefront theme when there is an additional express payment option (Google Pay or Apple Pay). It appears that the logo will scale down so that the words are still readable and in some cases (when the CTA is Donate with) the logo wasn't displayed at all. I believe this will require the three column layout on the block cart page that comes with out of the box Storefront theme.

### To Reproduce

1. Have a merchant store with WooPay and Google or Apple Pay enabled and displayed on the cart page.
2. For the Express checkout buttons turn on a CTA text (Buy with, Donate with, or Book with).
3. Have a merchant store with the Storefront theme that has three columns on the cart page.
4. Have a merchant store with a cart block page.
5. As a shopper add a product to the cart and visit the cart block page.
6. Observe that the WooPay logo in the button is either scaled down very small or not visible at all in the case of Donate with.

### Actual behavior

The WooPay logo displayed within the button is scaled down and barely visible in the above scenario.

### Screenshots

Screen Shot 2023-05-02 at 1 59 54 PM

### Expected behavior

The WooPay logo should stay scaled up and visible similar to how the Google Pay logo works. This appears to be achieved by making margins and text size smaller while making the logo only slightly smaller.

### PR DESCRIPTION
**Pull request by lovo-h**
* Fixes #6263 

#### Changes proposed in this Pull Request

This PR determines the width of the WooPay button upon render and depending on the width, it resizes the text and the logo size in order to fit within the button's area. Due to these changes being minor appearance changes, tests were withheld.

Here are various image examples of how this PR fixes the WooPay button in various contexts:
* Call to action: "Only icon", "Buy with", "Donate with", and "Book with".
* Button size: "Default (40 px)", "Medium (48 px)", and "Large (56 px)".

**Only Icon**

image

**Buy with**

image

**Donate with**

image

**Book with**

image

#### Testing instructions

* Ensure that WooPay is enabled for the merchant site.
* As a merchant, make sure to add a WCBlocks cart page and a WCBlocks checkout page.
* On `window A`, as a merchant, navigate to `WooCommerce > Settings > Payments`, click on `WooPayments` method, scroll down and click on `Customize` button for `WooPay`, under `Express checkouts` section.
* On a separate `window B`, as a customer, navigate to these 4 pages: `cart`, `cart with wc-blocks`, `checkout`, and `checkout with wc-blocks` making sure that the WooPay button's logo is not cut off.
* Repeat this for all of the WooPay "call to action" (4 of them) and "Button size" (3 of them) and ensure all buttons are rendered correctly.
* Repeat these tests on mobile and ensure all pages display the WooPay button correctly.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6939` or branch name `fix/woopay-6263-button-logo-cutoff` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/woopay-6263-button-logo-cutoff) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 92e048a135d6a9f5374d5d26bc3a1a06db65aab2
- Build time: 2023-08-09 13:11:51 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/checkout/woopay/express-button/woopay-express-checkout-button.js [hunk 1]
```diff
@@ -88,9 +110,10 @@ export const WoopayExpressCheckoutButton = ( {
 			data-type={ buttonType }
 			data-size={ size }
 			data-theme={ theme }
+			data-width-type={ buttonWidthType }
 			style={ { height: `${ height }px` } }
 		>
-			{ text }
+			<div>{ text }</div>
```
Comment (reykjalin) [inline client/checkout/woopay/express-button/woopay-express-checkout-button.js]: Is the `div` here necessary? It's heavily discouraged by the [HTML spec](https://html.spec.whatwg.org/multipage/form-elements.html#the-button-element) (see the "Content model" section) to put anything but [phrasing content](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content-2) into button tags.

If we need a containing element for the text, would a `span` suffice?

**Review by reykjalin — APPROVED** [review]
Tested the product, cart block, checkout block, cart shortcode, and checkout shortcode pages on desktop and my iPhone 13 mini and the changes look good. Code looks good too, just one small nitpick that I don't think should hold up the PR 🙂

File: client/checkout/woopay/express-button/woopay-express-checkout-button.js (hunk 1 above)
Comment (bborman22) [reply client/checkout/woopay/express-button/woopay-express-checkout-button.js]: Good catch @reykjalin! With @lovo-h AFK this week and in the interest of keeping this issue moving, I removed the div entirely and tested out the above scenarios with success! If you wouldn't mind giving it another look with that div removed before we merge this?

Comment (lovo-h) [reply client/checkout/woopay/express-button/woopay-express-checkout-button.js]: Good catch. Thank you both.

**Review by reykjalin — APPROVED** [review]
Looks good after @bborman22's changes 🚀

**Pull request merged**